tunnel/default.go

```diff
@@ -862,8 +862,11 @@ func (t *tun) setupLink(node string) (*link, error) {
 
 	// create a new link
 	link := newLink(c)
+
 	// set link id to remote side
+	link.Lock()
```

Comment:
You cannot lock here because setupLink is called by setupLinks which is called by Connect which also locks